protocols/relay/src/priv_client.rs

```diff
@@ -168,12 +168,14 @@ impl NetworkBehaviour for Behaviour {
         local_addr: &Multiaddr,
         remote_addr: &Multiaddr,
     ) -> Result<THandler<Self>, ConnectionDenied> {
+        let pending_handler_command = self.pending_handler_commands.remove(&connection_id);
+
         if local_addr.is_relayed() {
             return Ok(Either::Right(dummy::ConnectionHandler));
         }
         let mut handler = Handler::new(self.local_peer_id, peer, remote_addr.clone());
 
-        if let Some(event) = self.pending_handler_commands.remove(&connection_id) {
+        if let Some(event) = pending_handler_command {
```

Comment:
nit: I feel it was more readable the previous way as it was all on a single line. You mention that:
>However, it requires that in handle_established_{in, out}bound_connection the event is also removed when the connection is relayed, in which case the event is simply discarded because the commands are only valid on direct connections.

but this happened before right?

Reply:
> but this happened before right?

I am not sure what you mean. Could you rephrase? :)

We now need to remove the pending handler commands before the early return when `local_addr.is_relayed()`. That's why I moved it to the top of the function, and have to use the resulting `pending_handler_command` here.

Reply:
makes sense, thanks for explaining Elena